### PR TITLE
enable protocol/squash root

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-shares/test/ut/share_client_test.cpp [hunk 1]
```diff
@@ -523,14 +523,21 @@ namespace Azure { namespace Storage { namespace Test {
     auto shareName = m_testNameLowercase;
     auto shareClient = Files::Shares::ShareClient::CreateFromConnectionString(
         PremiumFileConnectionString(), shareName, m_options);
-    EXPECT_NO_THROW(shareClient.Create());
+    // create works
+    Files::Shares::CreateShareOptions options;
+    options.EnabledProtocols = Files::Shares::Models::ShareProtocols::Nfs;
+    options.RootSquash = Files::Shares::Models::ShareRootSquash::AllSquash;
+    EXPECT_NO_THROW(shareClient.Create(options));
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/test/ut/share_client_test.cpp:530]: Don't add tests like this. We don't know if the default options would work.

Create another share.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/test/ut/share_client_test.cpp:530]: Also need to get share properties and compare the value of `EnabledProptocols` and `RootSquash`.

Comment (microzchang) [reply sdk/storage/azure-storage-files-shares/test/ut/share_client_test.cpp]: fix it and add other two statement blocks to test nfs and smb protocols.

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**